File: app/src/main/java/io/moonshard/moonshard/models/roomEntities/MessageEntity.java
package io.moonshard.moonshard.models.roomEntities;


import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "messages")
public class MessageEntity {
    @PrimaryKey(autoGenerate = true) public long messageID; // Global message ID
    @ColumnInfo public String chatID; // Chat ID
    @ColumnInfo public String messageUid;
    @ColumnInfo public String senderJid;
    @ColumnInfo public long timestamp; // Timestamp
    @ColumnInfo public String text; // Message text
    @ColumnInfo public boolean isSent; // Send status indicator
    @ColumnInfo public boolean isRead; // Message Read Indicator

    public MessageEntity(String chatID, String messageUid, String senderJid, long timestamp, String text, boolean isSent, boolean isRead) {
        this.chatID = chatID;
        this.messageUid = messageUid;
        this.senderJid = senderJid;
        this.timestamp = timestamp;
        this.text = text;
        this.isSent = isSent;
        this.isRead = isRead;
    }

    @NonNull
    @Override
    public String toString() {
        return text;
    }
}

File: app/src/main/java/io/moonshard/moonshard/models/dbEntities/MessageEntity.kt
package io.moonshard.moonshard.models.dbEntities

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.Unique
import io.objectbox.relation.ToOne

@Entity
data class MessageEntity(
    @Id var id: Long = 0,
    @Unique var messageUid: String = "",
    @Unique var stanzaId: String? = null,
    var timestamp: Long = -1,
    var text: String = "",
    var isSent: Boolean = false,
    var isRead: Boolean = false,
    var isCurrentUserSender: Boolean = false,
    var isSystemMessage: Boolean = false,
    var isFile:Boolean=false
) {
    lateinit var chat: ToOne<ChatEntity>
    lateinit var sender: ToOne<ChatUser>
    
    override fun toString(): String {
        return text
    }
}